TypeCobol.LanguageServer/Workspace.cs

```diff
@@ -587,10 +587,23 @@ private void RefreshCustomSymbols()
         private void ProgramClassChanged(object cUnit, ProgramClassEvent programEvent)
         {
             var compilationUnit = cUnit as CompilationUnit;
-            var fileUri = OpenedDocumentContext.Keys.FirstOrDefault(k => compilationUnit != null && k.LocalPath.Contains(compilationUnit.TextSourceInfo.Name));
+            var fileUri = OpenedDocumentContext.Keys.FirstOrDefault(k =>
+                compilationUnit != null && k.LocalPath.Contains(compilationUnit.TextSourceInfo.Name));
 
-            var diags = compilationUnit?.AllDiagnostics().Take(TypeCobolConfiguration.MaximumDiagnostics == 0 ? 100 : TypeCobolConfiguration.MaximumDiagnostics);
-            DiagnosticsEvent(fileUri, new DiagnosticEvent() { Diagnostics = diags});
+            var test = compilationUnit?.AllDiagnostics().GroupBy(d => d.Info.Severity);
+            System.Collections.Generic.IEnumerable<Diagnostic> diags = new List<Diagnostic>();
+
+            var severityGroups = compilationUnit?.AllDiagnostics().GroupBy(d => d.Info.Severity);
```

Comment:
OrderBy doesn't work directly?

Reply:
OrderBy would work but the errors would be at the end of the list. We would need to reverse that list and change the order of appearance of the diagnostics. Since many of our LSR tests are already handling warnings and errors, they would be false.
The only way to change the priority of the errors without doing to much changes in the LSR test is to move up the errors as a group. In this way, the order of the diagnostics won't change within the groups.

Reply:
Ok, can you add this comment on the source ?